src/extensions/surveys/components/QuestionTypes.tsx

```diff
@@ -320,6 +342,12 @@ export function MultipleChoiceQuestion({
         if (question.type === SurveyQuestionType.SingleChoice) {
             setSelectedChoices(val)
             setOpenChoiceSelected(false) // Deselect open choice when selecting another option
+            if (shouldSkipSubmit) {
```

Comment:
I wonder how this would work if a survey has multiple questions, and some apply the `shouldSkipSubmit` and some do not. Does this mean we go automatically to the next question when `shouldSkipSubmit=true`?
What if it's a one-question survey? Do we automatically send the survey sent?

Reply:
> Does this mean we go automatically to the next question when `shouldSkipSubmit=true`?

if the question has the option, once clicked, we submit the response and go to the next question.

If the survey only has one question, it means two outcomes:

1. It goes to the confirmation message, if that exists
2. It closes the survey (as it's finished)

Regardless, in both cases the survey is submitted after clicking on the option

If the survey has multiple questions, some skipping the submit and some not, there's not a lot of difference. It means the questions with `shouldSkipSubmit=true` will not have the button and will submit on press, and the others won't. 

[here's a demo of these scenarios](https://www.loom.com/share/b04c76abee014fd883e65d27841a39bf)

Reply:
> if the question has the option, once clicked, we submit the response and go to the next question.

what do you mean by 'submit the response'? does that mean a single survey sent event with a single response?
how do we merge multiple survey sent events with multiple responses?

Reply:
no, not talking about the events here. Submit the response means executing the `onSubmit` method which is passed as a prop to the question component. Nothing to do with the `survey sent` events which is handled on another part of the code

Reply:
gotcha, understood.
iirc the feature request was mainly about making this even faster which means if its a single question, lets say a yes/no or thumbs up and down, you click on it and its done (survey dismissed - event sent), does that work this way as well if its a single question with skip next button enabled?

Reply:
i guess the way how it'd work now is only if you also skip the confirmation message?